viadot/sources/mindful.py

```diff
@@ -0,0 +1,251 @@
+import os
+import json
+import logging
+from io import StringIO
+from pydantic import BaseModel
+from requests.models import Response
+from datetime import datetime, timedelta
+from typing import Any, Dict, Tuple, Literal, Optional
+
+
+import pandas as pd
+
+from ..config import get_source_credentials
+from ..exceptions import CredentialError, APIError
+from ..utils import handle_api_response
+from .base import Source
+
+
+class MINDFUL_CREDENTIALS(BaseModel):
+    client_id: str
+    client_secret: str
+    customer_uuid: str
+    auth_token: str
+    vault: str
```

Comment:
Add docstrings to explain what these things mean

Reply:
added.